### ISSUE TITLE
Too many FOSS note taking software for Android: The devil's in the details

### ISSUE THREAD

**Issue by timofonic** [plafue/writeily-pro#159]
I would like to use a note taking software on both my Android eink device and smartphone, but I can't decide what to use. I like the features of many of them, but Im unable to decide to use only one app to keep chaos and organize everything.

I would love to see many of these features on one app and avoid proprietary stuff such as EverNote, but there's too many projects.

- @plafue 's (https://github.com/plafue/writeily-pro/)[Writeily Pro] has a nice interface and MarkDown support.
- @aykit 's (https://github.com/aykit/MyOwnNotes)[MyOwnNotes] supports OwnCloud and that's nice, but lacks the other nice features.
- @Groomiac 's (https://github.com/Groomiac/crocodilenote/)[CocodrileNote] provides encryption.
- @grote 's (https://github.com/grote/BlitzMail)[BlitzMail] has a nice email support.
- @/budowski 's (https://github.com/budowski/budoist)[Buboist] has some linkify and organizing features the rest lack, but depends on the Todoist proprietary service.
-  @thiagokimo 's (https://github.com/thiagokimo/Diktafon) and @/herverenault 's (https://github.com/herverenault/DirectDictaphone)[DirectDictaphone]has a nice voice note and speech recognition features.
- @atd 's (https://github.com/atd/HandyNotes)[HandyNotes] and https://github.com/MohammadAdib/FloatingStickies have sticky widgets for looking easily at them.
- @konradrenner 's (https://github.com/konradrenner/kolabnotes-android)[KolabNotes] has nice Kolab support.
- @spacecowboy 's (https://github.com/spacecowboy/NotePad)[NotePad] supports Google syncing.
@openintents ' (https://github.com/openintents/notepad)Notepad] has nice syncing features too.

As you devs see, there's lots of apps for note taking on Android and I just did a kick find other GitHub. I'm sure there's more FOSS note taking software for Android elsewhere too.

What's the barrier to join efforts in a common project? I see too many repeated efforts, yet they miss some nice details done by other apps.

(I wrote this message here because I'm tired, I think referencing people is enough to make others read this. I don't have a strong personal preference on certain software but: I must say Writeily Pro has a comfortable interface and MarkDown is a major killer app).

Kind regards.

**Comment by konradrenner** [comment]
@timofonic before I started developing KolabNotes I also searched for a sufficient note app, but I did not find any App which supports Kolab sync. So I would really love to see an OSS app which supports all those features which you mentioned.
If one takes the challenges and organize the implementation of such an app, I will contribute Kolab sync support. But this app has to support at least the features, which are supported by the Kolab Notes specification:
 - RichText (incl. inline images)
 - Notebooks
 - Attachments
 - Tags
 - Notebook sharing (incl. different permissions)
 - Colors for notes and tags

By the way the Kolab Notes app supports sticky widgets and list widgets ;-)

**Comment by thiagokimo** [comment]
Hi @timofonic,

Yes, I do often try thinking about non-common apps/ideas that I could work with in my spare time. Unfortunately is quite hard to find anything worth it. The **Diktafon** is one of the exemples. I simple stopped working on it because I found several voice/note recorders in Google Play.

One thing that could motivate me contributing with a project, despite its commonness, is the challenge of doing an unique UI for it (with animations, transitions and blablabla).

**Comment by Groomiac** [comment]
I did not care about UI and stuff when creating CrocodileNote, I just
needed simple text nodes with a directory. But I wanted it encrypted
bulletproofed. I have a friend who knows about hacking, security and
crypto because it's his daily job.
So with his help, I wrote the app. However, I would be willing to
participate/contribute for the security and possible crypto part for
something else.

I received some feature requests from users, but since I wrote it in my
spare time only, there was no reason to develop more features. I still
take care of the app running on the corresponding newest Android since I
am using the app. ;)

Simon (Groomiac)

timofonic:

**Comment by timofonic** [comment]
@konradrenner 

Thanks a lot for your fast reply!

Does "RichText" (it's an ambiguous definition for me, what do you mean) can be mapped to MarkDown?

What kind of attachments are we talking about? Could those be references to another system like Samba,  @syncthing 's [Block Exchange Protocol](http://docs.syncthing.net/specs/bep-v1.html)?

Can Kolab work on an average SMTP server even including mainstream ones such as Gmail?

**Comment by konradrenner** [comment]
@timofonic with Richtext I mean formatted text. It should be possible to format text at least with a GUI (eg a "Bold" button), if it is also possible to format text with markdown, that would be awesome, but I think it has to be stored as HTML Text, so that is parseable with a Kolab server or another Kolab client or some other client.

Example for attachments would be a voice note or a presentation  of a meeting or something like that. Kolab objects (eg a note) are IMAP messages which contain a xml file with a specified content and attachments are simple mail attachments.

With my Kolab Notes app it is already possible to sync with IMAP Server and so also with Gmail. There is no need for SMTP, a server just needs to support IMAP. But in case of Gmail, you don't have a webclient with which you can edit the note.

**Comment by pulimento** [comment]
@timofonic I totally agree with you. I'm not related to any app/developer, but I will be happy to contribute.

I haven't tried all the apps you've mentioned (only writeily and myownnotes). We can make THE NOTES APP if all colaborate a bit. Seeing the code, this app looks great for use as base.

I'm looking for a notes app with MD editing, image uploading, and owncloud sync, at least

**Comment by konradrenner** [comment]
Hi, is there any progress on this issue? @rwrnet also mentioned a nice possible feature in an issue for the Kolab Notes app (https://github.com/konradrenner/kolabnotes-android/issues/106).

If someone can do the coordination, I would be happy to do the Kolab support

**Comment by JasperWeiss** [comment]
For me, encryption is the most important feature besides the UI.
I'd love to see writely support Encryption. 
I'd be great if encryption was done with GPG using an asymmetric key. So notes can be decrypted on other platforms, too.

But AES256 using the lock password with some key stretching should be a good start.

**Comment by konradrenner** [comment]
I have reflected this issue, I think creating a new app, which can sync with many different providers on its own will never be finished. Also there will be problems with encryption (maybe some sync backend are not supporting it) and formatting of notes (markdow, html, etc.).

So my idea is:
We create a new app, which focuses on the UI an internal storage of data and we define how a Content Provider has to look, so that it works with this app. So it would be possible to fetch data from different already existing apps, which want to participate. The new notes app will internally handle the conversion from and to the content providers of the internal data structure.

What do you think about it?

**Comment by JasperWeiss** [comment]
@konradrenner That's a great idea. The entire internal data structure should be encrypted. The only thing the server gets to see is a blop of encrypted text. Should be compatible with virtually every sync service.
It should either be encrypted using AES256 with a passphrase (with some stretching) or with pgp using Openkeychain. Or allow the user to choose.

**Comment by jfkimmes** [comment]
Please do this! I would love to help, but I have no experience in Android development. 
If this really takes some shape I would love to test and translate to German?

**Comment by jkufner** [comment]
About the sync, I just started to use Writely with SGit. It looks like a promising combination. If these two get better integrated (open SGit directly from Writely, maybe automated commit&push), they could form a really powerful tool.

But to write ordinary short notes I use [OpenTasks](https://github.com/dmfs/opentasks) with a dedicated calendar synced to CalDAV server ([Davdroid](https://davdroid.bitfire.at)). Unfortunately not many apps support VJournal, but tasks work fine.

About the encryption: Calendar, contacts, sms, e-mails, ... full phone encryption is the only way. Don't waste your time here.
